adam_core/utils/indexable.py

```diff
@@ -399,6 +399,39 @@ def _query_index(
     def __len__(self):
         return len(self._index)
 
+    def __eq__(self, other: "Indexable") -> bool:
+        for k, v in self.__dict__.items():
```

Comment:
What if `other` has a key which `self` lacks?

One simple suggestion would be to check if `set(self.__dict__.keys()) == set(other.__dict__.keys())`.

Reply:
Excellent catch. Will update as you suggest.